### DISCUSSION TITLE
Close output file after writing the report

### DISCUSSION THREAD

**Discussion by roi-orca** [aquasecurity/trivy#4441] [Bugs]
## Description

When running trivy with --output flag and then our application try to delete the report file on windows, the deletion fail on `The process cannot access the file because it is being used by another process` since the file is not closed in trivy.

Reference:
https://github.com/aquasecurity/trivy/blob/main/pkg/flag/report_flags.go#L217-L222

## What did you expect to happen?
Close the file after writing the report

## What happened instead?
The file is not closed

**Comment by krishnaduttPanchagnula** [comment]
The os.Create truncates if the file exists. We can bypass this by checking if file exists, and if it does we can remove that file to create new file and output is stored in that file. What do you think @knqyf263 ?

 @roi-orca When i tried implemeted the file close in that file, it seems the file is already closed as i am getting errors in the integration tests.

**Reply by knqyf263** [reply]
What is the problem with truncating? And are you also seeing the error?

**Reply by roi-orca** [reply]
You can't close the file here since the reporter use the opened file, thus the reporter will need to open it again if closed here.
We can pass the reporter the file name and not the descriptor itself and handle the open and close by himself instead.

**Reply by krishnaduttPanchagnula** [reply]
> What is the problem with truncating? And are you also seeing the error?

By truncating we will be seeing the results of our previous runs and difficult to navigate through the whole doc.  I was not able to replicate this issue @knqyf263 .

**Comment by knqyf263** [comment]
Created an issue for tracking
https://github.com/aquasecurity/trivy/issues/4841
